engine/googletest_result_test.cpp

```diff
@@ -0,0 +1,431 @@
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "engine/googletest_result.hpp"
+
+extern "C" {
+#include <signal.h>
+}
+
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <stdexcept>
+
+#include <atf-c++.hpp>
+
+#include "engine/exceptions.hpp"
+#include "model/test_result.hpp"
+#include "utils/format/macros.hpp"
+#include "utils/fs/path.hpp"
+#include "utils/process/status.hpp"
+
+namespace fs = utils::fs;
+namespace process = utils::process;
+
+using utils::none;
+using utils::optional;
+
+
+namespace {
+
+
+/// Performs a test for results::parse() that should succeed.
+///
+/// \param exp_type The expected type of the result.
+/// \param exp_argument The expected argument in the result, if any.
+/// \param exp_reason The expected reason describing the result, if any.
+/// \param text The literal input to parse; can include multiple lines.
+static void
+parse_ok_test(const engine::googletest_result::types& exp_type,
+              const optional< int >& exp_argument,
+              const char* exp_reason, const char* text)
+{
+    std::istringstream input(text);
+    const engine::googletest_result actual =
+        engine::googletest_result::parse(input);
+    ATF_REQUIRE_EQ(exp_type, actual.type());
+    ATF_REQUIRE_EQ(exp_argument, actual.argument());
+    if (exp_reason != NULL) {
+        ATF_REQUIRE(actual.reason());
+        ATF_REQUIRE_EQ(exp_reason, actual.reason().get());
+    } else {
+        ATF_REQUIRE(!actual.reason());
+    }
+}
+
+
+/// Wrapper around parse_ok_test to define a test case.
+///
+/// \param name The name of the test case; will be prefixed with
+///     "googletest_result__parse__".
+/// \param exp_type The expected type of the result.
+/// \param exp_argument The expected argument in the result, if any.
+/// \param exp_reason The expected reason describing the result, if any.
+/// \param input The literal input to parse.
+#define PARSE(name, exp_type, exp_argument, exp_reason, input) \
+    ATF_TEST_CASE_WITHOUT_HEAD(googletest_result__parse__ ## name); \
+    ATF_TEST_CASE_BODY(googletest_result__parse__ ## name) \
+    { \
+        parse_ok_test(exp_type, exp_argument, exp_reason, input); \
+    }
+
+
+}  // anonymous namespace
+
+
+PARSE(broken,
+      engine::googletest_result::broken, none, "invalid output",
+      "invalid input");
+
+const char disabled_context[] = (
```

Comment:
nit: Same comment about other global strings in another test program. Just inline them?

Reply:
I'm pushing back on this: doing what you ask here will make things a lot less readable as this is all squeezed in to a macro (`PARSE(..)`).